# -*- coding: utf-8 -*-
# author: github.com/SelmanDedeakay
#
from PyQt5 import QtCore, QtGui, QtWidgets
import keyboard #for adding hotkeys
from textblob import TextBlob #for translating. It gets information from the Google Translate
from languages import * #It contains our language codes

class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow") #It's all about the design until the line 115
        MainWindow.resize(319, 830)
        MainWindow.setMinimumSize(QtCore.QSize(319, 830))
        MainWindow.setMaximumSize(QtCore.QSize(319, 830))
        icon = QtGui.QIcon()

        icon.addPixmap(QtGui.QPixmap(":/icons/1j+ojVVCOMkX9Wyrexe4hGe...png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        MainWindow.setWindowIcon(icon)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.label_logo = QtWidgets.QLabel(self.centralwidget)
        self.label_logo.setGeometry(QtCore.QRect(60, 10, 256, 114))
        self.label_logo.setText("")
        self.label_logo.setPixmap(QtGui.QPixmap(
            ":/icons/1j+ojVVCOMkX9Wyrexe4hGfNk5HZ+EAa3AiR1TN9Nmkd+AhlkSYpgfts8Ot+dERBpAQGghIOc8I0iCF6VJZE0kU7tmLSfMhOWg==.png"))
        self.label_logo.setObjectName("label_logo")
        self.layoutWidget = QtWidgets.QWidget(self.centralwidget)
        self.layoutWidget.setGeometry(QtCore.QRect(30, 130, 258, 666))
        self.layoutWidget.setObjectName("layoutWidget")
        self.formLayout = QtWidgets.QFormLayout(self.layoutWidget)
        self.formLayout.setContentsMargins(0, 0, 0, 0)
        self.formLayout.setObjectName("formLayout")
        self.label_menu = QtWidgets.QLabel(self.layoutWidget)
        self.label_menu.setObjectName("label_menu")
        self.formLayout.setWidget(0, QtWidgets.QFormLayout.LabelRole, self.label_menu)
        self.pushButton_english = QtWidgets.QPushButton(self.layoutWidget)
        self.pushButton_english.setObjectName("pushButton_english")
        self.formLayout.setWidget(0, QtWidgets.QFormLayout.FieldRole, self.pushButton_english)
        self.pushButton_turkish = QtWidgets.QPushButton(self.layoutWidget)
        self.pushButton_turkish.setObjectName("pushButton_turkish")
        self.formLayout.setWidget(1, QtWidgets.QFormLayout.FieldRole, self.pushButton_turkish)
        self.label_languageFrom = QtWidgets.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setFamily("-apple-system,.SF NS Text,Helvetica Neue,Lucida Grande")
        font.setPointSize(11)
        self.label_languageFrom.setFont(font)
        self.label_languageFrom.setObjectName("label_languageFrom")
        self.formLayout.setWidget(2, QtWidgets.QFormLayout.SpanningRole, self.label_languageFrom)
        self.comboBox_languageFrom = QtWidgets.QComboBox(self.layoutWidget)
        self.comboBox_languageFrom.setMaxVisibleItems(8)
        self.comboBox_languageFrom.setObjectName("comboBox_languageFrom")
        self.comboBox_languageFrom.addItem("")
        self.comboBox_languageFrom.addItem("")
        self.comboBox_languageFrom.addItem("")
        self.comboBox_languageFrom.addItem("")
        self.comboBox_languageFrom.addItem("")
        self.comboBox_languageFrom.addItem("")
        self.comboBox_languageFrom.addItem("")
        self.comboBox_languageFrom.addItem("")
        self.comboBox_languageFrom.addItem("")
        self.comboBox_languageFrom.addItem("")
        self.comboBox_languageFrom.addItem("")
        self.comboBox_languageFrom.addItem("")
        self.comboBox_languageFrom.addItem("")
        self.formLayout.setWidget(3, QtWidgets.QFormLayout.SpanningRole, self.comboBox_languageFrom)
        self.label_languageTo = QtWidgets.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(11)
        self.label_languageTo.setFont(font)
        self.label_languageTo.setObjectName("label_languageTo")
        self.formLayout.setWidget(4, QtWidgets.QFormLayout.LabelRole, self.label_languageTo)
        self.comboBox_languageTo = QtWidgets.QComboBox(self.layoutWidget)
        self.comboBox_languageTo.setMaxVisibleItems(8)
        self.comboBox_languageTo.setObjectName("comboBox_languageTo")
        self.comboBox_languageTo.addItem("")
        self.comboBox_languageTo.addItem("")
        self.comboBox_languageTo.addItem("")
        self.comboBox_languageTo.addItem("")
        self.comboBox_languageTo.addItem("")
        self.comboBox_languageTo.addItem("")
        self.comboBox_languageTo.addItem("")
        self.comboBox_languageTo.addItem("")
        self.comboBox_languageTo.addItem("")
        self.comboBox_languageTo.addItem("")
        self.comboBox_languageTo.addItem("")
        self.comboBox_languageTo.addItem("")
        self.formLayout.setWidget(5, QtWidgets.QFormLayout.SpanningRole, self.comboBox_languageTo)
        self.textEdit_text = QtWidgets.QTextEdit(self.layoutWidget)
        self.textEdit_text.setObjectName("textEdit_text")
        self.formLayout.setWidget(6, QtWidgets.QFormLayout.SpanningRole, self.textEdit_text)
        self.textEdit_translated = QtWidgets.QTextEdit(self.layoutWidget)
        self.textEdit_translated.setObjectName("textEdit_translated")
        self.formLayout.setWidget(8, QtWidgets.QFormLayout.SpanningRole, self.textEdit_translated)
        self.pushButton_translate = QtWidgets.QPushButton(self.layoutWidget)
        self.pushButton_translate.setObjectName("pushButton_transtlate")
        self.formLayout.setWidget(9, QtWidgets.QFormLayout.SpanningRole, self.pushButton_translate)
        self.textEdit_translated.raise_()
        self.label_menu.raise_()
        self.pushButton_english.raise_()
        self.pushButton_turkish.raise_()
        self.label_languageFrom.raise_()
        self.comboBox_languageFrom.raise_()
        self.label_languageTo.raise_()
        self.comboBox_languageTo.raise_()
        self.textEdit_text.raise_()
        self.pushButton_translate.raise_()
        MainWindow.setCentralWidget(self.centralwidget)
        self.statusbar = QtWidgets.QStatusBar(MainWindow)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)
        self.actionEng = QtWidgets.QAction(MainWindow)
        self.actionEng.setObjectName("actionEng")
        self.retranslateUi(MainWindow)
        self.comboBox_languageFrom.setCurrentIndex(0)
        self.comboBox_languageTo.setCurrentIndex(10)
        self.menulang = "TR" #Basicly the default display language is Turkish.
        keyboard.add_hotkey('enter', lambda: self.pushButton_translate.click()) #It does the translate when pressed "ENTER"  on keyboard as a hotkey
        self.pushButton_translate.clicked.connect(self.translate) #It runs the translate function when clicked.

        self.pushButton_english.clicked.connect(self.changeMenuENG)
        self.pushButton_turkish.clicked.connect(self.changeMenuTR)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def translate(self):
        try:
            self.text = TextBlob(self.textEdit_text.toPlainText())
            langfrom = languagefrom(self.comboBox_languageFrom.currentIndex())
            langto = languageto(self.comboBox_languageTo.currentIndex())
            if langfrom == -1:
                self.text.detect_language()
                self.textEdit_translated.setPlainText(str(self.text.translate(to=langto)))
            else:
                self.textEdit_translated.setPlainText(str(self.text.translate(from_lang=langfrom,to=langto)))
        except:
           if self.menulang =="TR":
               self.textEdit_translated.setPlainText("Çeviri Hatası! Lütfen Metninizi Kontrol Edin!")
           else:
               self.textEdit_translated.setPlainText("Translation Error! Please Check Your Text!")
    def changeMenuENG(self):
        self.label_languageFrom.setText("From:")
        self.label_languageTo.setText("To:")
        self.textEdit_text.setPlaceholderText("Enter the text to be translated:")
        self.comboBox_languageFrom.setItemText(0, "Detect The Language")
        self.comboBox_languageFrom.setItemText(1, "German")
        self.comboBox_languageFrom.setItemText(2, "Arabic")
        self.comboBox_languageFrom.setItemText(3, "Bulgarian")
        self.comboBox_languageFrom.setItemText(4, "French")
        self.comboBox_languageFrom.setItemText(5, "English")
        self.comboBox_languageFrom.setItemText(6, "Spanish")
        self.comboBox_languageFrom.setItemText(7, "Italian")
        self.comboBox_languageFrom.setItemText(8, "Japanese")
        self.comboBox_languageFrom.setItemText(9, "Portuguese")
        self.comboBox_languageFrom.setItemText(10, "Russian")
        self.comboBox_languageFrom.setItemText(11, "Turkish")
        self.comboBox_languageFrom.setItemText(12, "Greek")
        self.comboBox_languageTo.setItemText(0, "German")
        self.comboBox_languageTo.setItemText(1, "Arabic")
        self.comboBox_languageTo.setItemText(2, "Bulgarian")
        self.comboBox_languageTo.setItemText(3, "French")
        self.comboBox_languageTo.setItemText(4, "English")
        self.comboBox_languageTo.setItemText(5, "Spanish")
        self.comboBox_languageTo.setItemText(6, "Italian")
        self.comboBox_languageTo.setItemText(7, "Japanese")
        self.comboBox_languageTo.setItemText(8, "Portuguese")
        self.comboBox_languageTo.setItemText(9, "Russian")
        self.comboBox_languageTo.setItemText(10, "Turkish")
        self.comboBox_languageTo.setItemText(11, "Greek")
        self.pushButton_translate.setText("Translate")
        self.label_menu.setText("Menu Language:")
        self.comboBox_languageTo.setCurrentIndex(4)
        self.menulang = "ENG"
    def changeMenuTR(self):
        self.label_languageFrom.setText("Çeviri Yapılacak Dil:")
        self.label_languageTo.setText("Çevirilecek Dil:")
        self.textEdit_text.setPlaceholderText("Çevirilecek Metni Giriniz:")
        self.comboBox_languageFrom.setItemText(0, "Dili Algıla")
        self.comboBox_languageFrom.setItemText(1, "Almanca")
        self.comboBox_languageFrom.setItemText(2, "Arapça")
        self.comboBox_languageFrom.setItemText(3, "Bulgarca")
        self.comboBox_languageFrom.setItemText(4, "Fransızca")
        self.comboBox_languageFrom.setItemText(5, "İngilizce")
        self.comboBox_languageFrom.setItemText(6, "İspanyolca")
        self.comboBox_languageFrom.setItemText(7, "İtalyanca")
        self.comboBox_languageFrom.setItemText(8, "Japonca")
        self.comboBox_languageFrom.setItemText(9, "Portekizce")
        self.comboBox_languageFrom.setItemText(10, "Rusça")
        self.comboBox_languageFrom.setItemText(11, "Türkçe")
        self.comboBox_languageFrom.setItemText(12, "Yunanca")
        self.comboBox_languageTo.setItemText(0, "Almanca")
        self.comboBox_languageTo.setItemText(1, "Arapça")
        self.comboBox_languageTo.setItemText(2, "Bulgarca")
        self.comboBox_languageTo.setItemText(3, "Fransızca")
        self.comboBox_languageTo.setItemText(4, "İngilizce")
        self.comboBox_languageTo.setItemText(5, "İspanyolca")
        self.comboBox_languageTo.setItemText(6, "İtalyanca")
        self.comboBox_languageTo.setItemText(7, "Japonca")
        self.comboBox_languageTo.setItemText(8, "Portekizce")
        self.comboBox_languageTo.setItemText(9, "Rusça")
        self.comboBox_languageTo.setItemText(10, "Türkçe")
        self.comboBox_languageTo.setItemText(11, "Yunanca")
        self.pushButton_translate.setText("Çevir")
        self.label_menu.setText("Menü Dili:")
        self.comboBox_languageTo.setCurrentIndex(10)
        self.menulang = "TR"
    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "T-Widget"))
        self.label_menu.setText(_translate("MainWindow", "Menü Dili:"))
        self.pushButton_english.setText(_translate("MainWindow", "ENG"))
        self.pushButton_turkish.setText(_translate("MainWindow", "TR"))
        self.label_languageFrom.setText(_translate("MainWindow", "Çeviri Yapılacak Dil:"))
        self.comboBox_languageFrom.setItemText(0, _translate("MainWindow", "Dili Algıla"))
        self.comboBox_languageFrom.setItemText(1, _translate("MainWindow", "Almanca"))
        self.comboBox_languageFrom.setItemText(2, _translate("MainWindow", "Arapça"))
        self.comboBox_languageFrom.setItemText(3, _translate("MainWindow", "Bulgarca"))
        self.comboBox_languageFrom.setItemText(4, _translate("MainWindow", "Fransızca"))
        self.comboBox_languageFrom.setItemText(5, _translate("MainWindow", "İngilizce"))
        self.comboBox_languageFrom.setItemText(6, _translate("MainWindow", "İspanyolca"))
        self.comboBox_languageFrom.setItemText(7, _translate("MainWindow", "İtalyanca"))
        self.comboBox_languageFrom.setItemText(8, _translate("MainWindow", "Japonca"))
        self.comboBox_languageFrom.setItemText(9, _translate("MainWindow", "Portekizce"))
        self.comboBox_languageFrom.setItemText(10, _translate("MainWindow", "Rusça"))
        self.comboBox_languageFrom.setItemText(11, _translate("MainWindow", "Türkçe"))
        self.comboBox_languageFrom.setItemText(12, _translate("MainWindow", "Yunanca"))
        self.label_languageTo.setText(_translate("MainWindow", "Çevirilecek Dil:"))
        self.comboBox_languageTo.setItemText(0, _translate("MainWindow", "Almanca"))
        self.comboBox_languageTo.setItemText(1, _translate("MainWindow", "Arapça"))
        self.comboBox_languageTo.setItemText(2, _translate("MainWindow", "Bulgarca"))
        self.comboBox_languageTo.setItemText(3, _translate("MainWindow", "Fransızca"))
        self.comboBox_languageTo.setItemText(4, _translate("MainWindow", "İngilizce"))
        self.comboBox_languageTo.setItemText(5, _translate("MainWindow", "İspanyolca"))
        self.comboBox_languageTo.setItemText(6, _translate("MainWindow", "İtalyanca"))
        self.comboBox_languageTo.setItemText(7, _translate("MainWindow", "Japonca"))
        self.comboBox_languageTo.setItemText(8, _translate("MainWindow", "Portekizce"))
        self.comboBox_languageTo.setItemText(9, _translate("MainWindow", "Rusça"))
        self.comboBox_languageTo.setItemText(10, _translate("MainWindow", "Türkçe"))
        self.comboBox_languageTo.setItemText(11, _translate("MainWindow", "Yunanca"))
        self.textEdit_text.setPlaceholderText(_translate("MainWindow", "Çevirilecek Metni Giriniz:"))
        self.pushButton_translate.setText(_translate("MainWindow", "Çevir"))


import icons_rc
